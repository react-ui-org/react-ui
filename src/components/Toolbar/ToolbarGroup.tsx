import React from 'react';
import { withGlobalProps } from '../../providers/globalProps';
import { classNames } from '../../helpers/classNames/classNames';
import { transferProps } from '../../helpers/transferProps';
import { isChildrenEmpty } from '../_helpers/isChildrenEmpty';
import { getAlignClassName } from './_helpers/getAlignClassName';
import { ToolbarGroupProps } from './ToolbarGroup.types';
import styles from './Toolbar.module.scss';

export const ToolbarGroup: React.FunctionComponent<ToolbarGroupProps> = ({
  align = 'top',
  children,
  dense = false,
  nowrap = false,
  ...restProps
}) => {
  if (isChildrenEmpty(children)) {
    return null;
  }

  return (
    <div
      {...transferProps(restProps)}
      className={classNames(
        styles.group,
        dense && styles.isGroupDense,
        nowrap && styles.isGroupNowrap,
        getAlignClassName(align, styles, 'group'),
      )}
    >
      {children}
    </div>
  );
};

export const ToolbarGroupWithGlobalProps = withGlobalProps(ToolbarGroup, 'ToolbarGroup');

export default ToolbarGroupWithGlobalProps;
