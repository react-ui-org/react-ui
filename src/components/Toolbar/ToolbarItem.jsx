import PropTypes from 'prop-types';
import React from 'react';
import { classNames } from '../../helpers/classNames/classNames';
import { transferProps } from '../../helpers/transferProps';
import { withGlobalProps } from '../../providers/globalProps';
import { isChildrenEmpty } from '../../helpers/isChildrenEmpty/isChildrenEmpty';
import styles from './Toolbar.module.scss';

export const ToolbarItem = ({
  children,
  flexible,
  ...restProps
}) => {
  if (isChildrenEmpty(children)) {
    return null;
  }

  return (
    <div
      {...transferProps(restProps)}
      className={classNames(
        styles.item,
        flexible && styles.isItemFlexible,
      )}
    >
      {children}
    </div>
  );
};

ToolbarItem.defaultProps = {
  children: undefined,
  flexible: false,
};

ToolbarItem.propTypes = {
  /**
   * Content of the toolbar item. If none are provided nothing is rendered.
   */
  children: PropTypes.node,
  /**
   * Allow item to grow and shrink if needed.
   */
  flexible: PropTypes.bool,
};

export const ToolbarItemWithGlobalProps = withGlobalProps(ToolbarItem, 'ToolbarItem');

export default ToolbarItemWithGlobalProps;
