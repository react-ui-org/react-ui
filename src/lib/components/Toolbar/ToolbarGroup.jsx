import PropTypes from 'prop-types';
import React from 'react';
import { withProviderContext } from '../../provider';
import { classNames } from '../../utils/classNames';
import styles from './Toolbar.scss';

export const ToolbarGroup = (props) => {
  const {
    align,
    children,
    dense,
    nowrap,
  } = props;

  const alignClass = (value) => {
    if (value === 'top') {
      return styles.isAlignedToTop;
    }

    if (value === 'middle') {
      return styles.isAlignedToMiddle;
    }

    if (value === 'bottom') {
      return styles.isAlignedToBottom;
    }

    return styles.isAlignedToBaseline;
  };

  return (
    <div
      className={classNames(
        styles.group,
        dense && styles.isDense,
        nowrap && styles.isNowrap,
        alignClass(align),
      )}
    >
      {children}
    </div>
  );
};

ToolbarGroup.defaultProps = {
  align: 'top',
  dense: false,
  nowrap: false,
};

ToolbarGroup.propTypes = {
  /**
   * Vertical alignment of toolbar items in the group.
   */
  align: PropTypes.oneOf(['top', 'middle', 'bottom', 'baseline']),
  /**
   * Grouped ToolbarItems.
   */
  children: PropTypes.node.isRequired,
  /**
   * If `true`, spacing of toolbar items in the group will be reduced.
   */
  dense: PropTypes.bool,
  /**
   * If set, the toolbar group will not wrap.
   */
  nowrap: PropTypes.bool,
};

export const ToolbarGroupWithContext = withProviderContext(ToolbarGroup, 'ToolbarGroup');

export default ToolbarGroupWithContext;
