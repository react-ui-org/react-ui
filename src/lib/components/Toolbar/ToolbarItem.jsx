import PropTypes from 'prop-types';
import React from 'react';
import { withProviderContext } from '../../provider';
import styles from './Toolbar.scss';

export const ToolbarItem = (props) => {
  const {
    children,
  } = props;

  return (
    <div className={styles.item}>
      {children}
    </div>
  );
};

ToolbarItem.propTypes = {
  /**
   * Content of the toolbar item.
   */
  children: PropTypes.node.isRequired,
};

export const ToolbarItemWithContext = withProviderContext(ToolbarItem, 'ToolbarItem');

export default ToolbarItemWithContext;
