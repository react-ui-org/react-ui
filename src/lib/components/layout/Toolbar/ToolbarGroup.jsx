import PropTypes from 'prop-types';
import React from 'react';
import styles from './Toolbar.scss';

export const ToolbarGroup = (props) => {
  const {
    align,
    children,
    dense,
    nowrap,
  } = props;

  const alignClass = (value) => {
    if (value === 'top') {
      return styles.isAlignedToTop;
    }

    if (value === 'middle') {
      return styles.isAlignedToMiddle;
    }

    if (value === 'bottom') {
      return styles.isAlignedToBottom;
    }

    return styles.isAlignedToBaseline;
  };

  return (
    <div
      className={[
        styles.group,
        dense ? styles.isDense : null,
        nowrap ? styles.isNowrap : null,
        alignClass(align),
      ].join(' ')}
    >
      {children}
    </div>
  );
};

ToolbarGroup.defaultProps = {
  align: 'top',
  dense: false,
  nowrap: false,
};

ToolbarGroup.propTypes = {
  /**
   * Vertical alignment of toolbar items in the group.
   */
  align: PropTypes.oneOf(['top', 'middle', 'bottom', 'baseline']),
  /**
   * Grouped ToolbarItems.
   */
  children: PropTypes.node.isRequired,
  /**
   * If `true`, spacing of toolbar items in the group will be reduced.
   */
  dense: PropTypes.bool,
  /**
   * If set, the toolbar group will not wrap.
   */
  nowrap: PropTypes.bool,
};

export default ToolbarGroup;
