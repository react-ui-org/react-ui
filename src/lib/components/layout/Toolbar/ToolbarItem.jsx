import PropTypes from 'prop-types';
import React from 'react';
import styles from './Toolbar.scss';

export const ToolbarItem = (props) => {
  const {
    children,
  } = props;

  return (
    <div className={styles.item}>
      {children}
    </div>
  );
};

ToolbarItem.propTypes = {
  /**
   * Content of the toolbar item.
   */
  children: PropTypes.node.isRequired,
};

export default ToolbarItem;
