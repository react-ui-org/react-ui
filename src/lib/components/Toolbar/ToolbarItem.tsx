import React from 'react';
import { transferProps } from '../_helpers/transferProps';
import { classNames } from '../../utils/classNames';
import { withGlobalProps } from '../../provider';
import { isChildrenEmpty } from '../_helpers/isChildrenEmpty';
import styles from './Toolbar.scss';
import { ToolbarItemProps } from './Toolbar.types';

export const ToolbarItem: React.FC<ToolbarItemProps> = ({
  children = null,
  flexible = false,
  ...restProps
}: ToolbarItemProps) => {
  if (isChildrenEmpty(children)) {
    return null;
  }

  return (
    <div
      {...transferProps(restProps)}
      className={classNames(
        styles.item,
        flexible && styles.isItemFlexible,
      )}
    >
      {children}
    </div>
  );
};

export const ToolbarItemWithGlobalProps = withGlobalProps(ToolbarItem, 'ToolbarItem');

export default ToolbarItemWithGlobalProps;
