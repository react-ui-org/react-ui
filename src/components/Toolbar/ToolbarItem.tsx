import React from 'react';
import { classNames } from '../../helpers/classNames/classNames';
import { transferProps } from '../../helpers/transferProps';
import { withGlobalProps } from '../../providers/globalProps';
import { isChildrenEmpty } from '../_helpers/isChildrenEmpty';
import { ToolbarItemProps } from './ToolbarItem.types';
import styles from './Toolbar.module.scss';

export const ToolbarItem: React.FunctionComponent<ToolbarItemProps> = ({
  children,
  flexible = false,
  ...restProps
}) => {
  if (isChildrenEmpty(children)) {
    return null;
  }

  return (
    <div
      {...transferProps(restProps)}
      className={classNames(
        styles.item,
        flexible && styles.isItemFlexible,
      )}
    >
      {children}
    </div>
  );
};

export const ToolbarItemWithGlobalProps = withGlobalProps(ToolbarItem, 'ToolbarItem');

export default ToolbarItemWithGlobalProps;
