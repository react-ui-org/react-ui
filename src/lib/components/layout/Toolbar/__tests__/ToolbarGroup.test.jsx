import React from 'react';
import { shallow } from 'enzyme';
import ToolbarGroup from '../ToolbarGroup';
import ToolbarItem from '../ToolbarItem';

describe('rendering', () => {
  it('renders correctly with a single child', () => {
    const tree = shallow((
      <ToolbarGroup>
        <ToolbarItem>item</ToolbarItem>
      </ToolbarGroup>
    ));

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with multiple children', () => {
    const tree = shallow((
      <ToolbarGroup>
        <ToolbarItem>item 1</ToolbarItem>
        <ToolbarItem>item 2</ToolbarItem>
        <ToolbarItem>item 3</ToolbarItem>
      </ToolbarGroup>
    ));

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with all props', () => {
    const tree = shallow((
      <ToolbarGroup align="middle" dense nowrap>
        <ToolbarItem>item</ToolbarItem>
      </ToolbarGroup>
    ));

    expect(tree).toMatchSnapshot();
  });
});
