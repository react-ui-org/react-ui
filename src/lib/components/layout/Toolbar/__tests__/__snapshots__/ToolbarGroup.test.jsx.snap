// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly with a single child 1`] = `
<div
  className="group isAlignedToTop"
>
  <ToolbarItem>
    item
  </ToolbarItem>
</div>
`;

exports[`rendering renders correctly with all props 1`] = `
<div
  className="group isAlignedToMiddle"
>
  <ToolbarItem>
    item
  </ToolbarItem>
</div>
`;

exports[`rendering renders correctly with multiple children 1`] = `
<div
  className="group isAlignedToTop"
>
  <ToolbarItem>
    item 1
  </ToolbarItem>
  <ToolbarItem>
    item 2
  </ToolbarItem>
  <ToolbarItem>
    item 3
  </ToolbarItem>
</div>
`;
